src/main/java/com/minecolonies/apiimp/initializer/ModBlocksInitializer.java

```diff
@@ -247,4 +250,14 @@ public static void registerBlockItem(final IForgeRegistry<Item> registry)
         ModBlocks.blockMediumQuarry.registerBlockItem(registry, new Item.Properties());
         //ModBlocks.blockLargeQuarry.registerBlockItem(registry, new Item.Properties());
     }
+
+    private static void registerCompostItems()
+    {
+        for (final Block block : ModBlocks.getCrops())
+        {
+            ComposterBlock.COMPOSTABLES.put(block.asItem(), 0.5f);
+        }
+
+        ComposterBlock.COMPOSTABLES.put(ModBlocks.blockCompostedDirt.asItem(), 1f);
```

Comment:
Not sure if it really makes sense to compost composted dirt which is already created by the composter, the AI could probably run into trouble here

Reply:
The composter doesn't make composted dirt.  They make compost OR dirt.  Composted dirt is a crafting table recipe which can be taught to the farmer.

Reply:
How does it check for excess? I recall that is one of the common issues that those cycles do things player did not intend

Reply:
There is no cycle.  There is no check.  It's up to the player to decide what they want to compost.  (And the mcol composter won't take items that aren't designated, or items that would reduce below the limits set in the warehouse.)